packages/eslint-plugin-svelte/src/rules/prefer-svelte-reactivity.ts

```diff
@@ -0,0 +1,285 @@
+							node
+						});
+					}
+					if (
+						path[0] === 'URLSearchParams' &&
+						isURLSearchParamsMutable(referenceTracker, node as TSESTree.Expression)
+					) {
+						context.report({
+							messageId: 'mutableURLSearchParamsUsed',
+							node
+						});
+					}
+				}
+			}
+		};
+	}
+});
+
+function isDateMutable(referenceTracker: ReferenceTracker, ctorNode: TSESTree.Expression): boolean {
+	return !referenceTracker
+		.iteratePropertyReferences(ctorNode, {
+			setDate: {
+				[ReferenceTracker.CALL]: true
+			},
+			setFullYear: {
+				[ReferenceTracker.CALL]: true
+			},
+			setHours: {
+				[ReferenceTracker.CALL]: true
+			},
+			setMilliseconds: {
+				[ReferenceTracker.CALL]: true
+			},
+			setMinutes: {
+				[ReferenceTracker.CALL]: true
+			},
+			setMonth: {
+				[ReferenceTracker.CALL]: true
+			},
+			setSeconds: {
+				[ReferenceTracker.CALL]: true
+			},
+			setTime: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCDate: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCFullYear: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCHours: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCMilliseconds: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCMinutes: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCMonth: {
+				[ReferenceTracker.CALL]: true
+			},
+			setUTCSeconds: {
+				[ReferenceTracker.CALL]: true
+			},
+			setYear: {
+				[ReferenceTracker.CALL]: true
+			}
+		})
+		.next().done;
+}
+
+function isMapMutable(referenceTracker: ReferenceTracker, ctorNode: TSESTree.Expression): boolean {
+	return !referenceTracker
+		.iteratePropertyReferences(ctorNode, {
+			clear: {
+				[ReferenceTracker.CALL]: true
+			},
+			delete: {
+				[ReferenceTracker.CALL]: true
+			},
+			set: {
+				[ReferenceTracker.CALL]: true
+			}
+		})
+		.next().done;
+}
+
+function isSetMutable(referenceTracker: ReferenceTracker, ctorNode: TSESTree.Expression): boolean {
+	return !referenceTracker
+		.iteratePropertyReferences(ctorNode, {
+			add: {
+				[ReferenceTracker.CALL]: true
+			},
+			clear: {
+				[ReferenceTracker.CALL]: true
+			},
+			delete: {
+				[ReferenceTracker.CALL]: true
+			}
+		})
+		.next().done;
+}
+
+function isURLMutable(referenceTracker: ReferenceTracker, ctorNode: TSESTree.Expression): boolean {
+	for (const { node } of referenceTracker.iteratePropertyReferences(ctorNode, {
+		hash: {
+			[ReferenceTracker.READ]: true
+		},
+		host: {
+			[ReferenceTracker.READ]: true
+		},
+		hostname: {
+			[ReferenceTracker.READ]: true
+		},
+		href: {
+			[ReferenceTracker.READ]: true
+		},
+		password: {
+			[ReferenceTracker.READ]: true
+		},
+		pathname: {
+			[ReferenceTracker.READ]: true
+		},
+		port: {
+			[ReferenceTracker.READ]: true
+		},
+		protocol: {
+			[ReferenceTracker.READ]: true
+		},
+		search: {
+			[ReferenceTracker.READ]: true
+		},
+		username: {
+			[ReferenceTracker.READ]: true
+		}
+	})) {
+		if (node.parent.type === 'AssignmentExpression') {
```

Comment:
I think we need to check if the node on the left.

```suggestion
		if (node.parent.type === 'AssignmentExpression' && node.parent.left === node) {
```

If we don't check that, I think it will match `x = url.hash`.